docs/linux/program-type/BPF_PROG_TYPE_KPROBE.md

```diff
@@ -87,11 +87,72 @@ The kprobe can be temporality disabled with the `PERF_EVENT_IOC_DISABLE` ioctl o
 
 ### Link kprobe attaching
 
-The more modern and preferred way of attaching is using the [link create command](../syscall/BPF_LINK_CREATE.md) of the BPF syscall.  
+The more modern and preferred way of attaching is using the [link create command](../syscall/BPF_LINK_CREATE.md) of the BPF syscall. 
+For single probes, open a new perf event using the [`perf_event_open`](https://man7.org/linux/man-pages/man2/perf_event_open.2.html) syscall. Note that the values of the attributes of the perf event structure are a little different here compared to the legacy way. 
 
 <!-- TODO First use PMU instead of DebugFS -->
+
+```c
+ struct perf_event_attr attr = {
+    .type = 8; /* read type from /sys/bus/event_source/devices/kprobe/type or uprobe/type */
+    .sample_type = PERF_SAMPLE_RAW;
+    .sample_period = 1;
+    .wakeup_events = 1;
+    .size = sizeof(attr);
+    .config |= 1 << 0;
+    .kprobe_func = ((uint64_t)"symbol_name");  /* symbol name in string, valid names can be found in /proc/kallsyms */
+    .config1 = ((uint64_t)"symbol_name"); 
+    .config2 = 0x0;
+    .probe_offset = 0x0; /* offset must be a valid instruction, here it is just the start of the kernel symbol*/
+};
+
+syscall(SYS_perf_event_open, 
+    &attr,  /* struct perf_event_attr * */
+    -1,     /* pid_t pid */
+    0       /* int cpu */
+    -1,     /* int group_fd */
+    PERF_FLAG_FD_CLOEXEC /* unsigned long flags */
+);
+```
 <!-- TODO then use Link instead of perf_event -->
 
+After the perf event syscall is successful, the valid file descriptor returned can be used to set the link_create.target_fd attribute in the bpf structure before the [link create command](../syscall/BPF_LINK_CREATE.md) is called.
+
+```c
+union bpf_attr attr = {
+    .link_create.prog_fd = prog_fd; /* valid fd to bpf program of type KPROBE */
+    .link_create.target_fd = perf_fd; /* valid fd to PMU event */
+    .link_create.attach_type = BPF_PERF_EVENT;
+    .link_create.flags = 0;
+    .link_create.perf_event.bpf_cookie = 0;
+};
+
+syscall(SYS_bpf,
+    BPF_LINK_CREATE,
+    &attr,
+    sizeof(attr)
+);
+```   
+For multiple probes, [link create command](../syscall/BPF_LINK_CREATE.md) can be used to combine the creation and linking of the probes. [Fprobes](https://lore.kernel.org/bpf/20220316122419.933957-4-jolsa@kernel.org/) are used under the hood for multiple kprobes.
```

Comment:
The spellcheck isn't happy about "Fprobes" as plural, please add it to the spellcheck dictionary ([.aspell.en.pws](https://github.com/isovalent/ebpf-docs/blob/master/.aspell.en.pws#L47))